File: src/main/java/nl/dannyjelsma/soundboardbot/commands/music/RemoveCommand.java
package nl.dannyjelsma.soundboardbot.commands.music;

import com.jagrosh.jdautilities.command.Command;
import com.jagrosh.jdautilities.command.CommandEvent;
import nl.dannyjelsma.soundboardbot.audio.SoundManager;
import nl.dannyjelsma.soundboardbot.audio.SoundScheduler;

public class RemoveCommand extends Command {

    public RemoveCommand() {
        name = "remove";
        help = "=remove <queue index>";
    }

    @Override
    protected void execute(CommandEvent event) {

        SoundManager manager = SoundManager.getInstance();
        SoundScheduler scheduler = manager.getScheduler();
        String[] args = event.getArgs().split(" ");

        if (args.length != 1) {
            event.getChannel().sendMessage("Invalid usage: =remove <queue index>").queue();
            return;
        }

        String indexStr = args[0];

        try {
            int index = Integer.parseInt(indexStr);

            scheduler.removeAtIndex(index);
            event.getChannel().sendMessage("Successfully removed the song from the queue!").queue();
        } catch (NumberFormatException ex) {
            event.getChannel().sendMessage(indexStr + " is not a number!").queue();
        }
    }
}

File: src/main/java/nl/dannyjelsma/soundboardbot/commands/music/RemoveCommand.kt
package nl.dannyjelsma.soundboardbot.commands.music

import com.jagrosh.jdautilities.command.Command
import com.jagrosh.jdautilities.command.CommandEvent
import nl.dannyjelsma.soundboardbot.audio.SoundManager

class RemoveCommand : Command() {
    init {
        name = "remove"
        help = "=remove <queue index>"
    }

    override fun execute(event: CommandEvent) {
        val args = event.args.split(" ").toTypedArray()
        if (args.size != 1) {
            event.channel.sendMessage("Invalid usage: =remove <queue index>").queue()
            return
        }

        val indexStr = args[0]
        try {
            val index = indexStr.toInt()
            SoundManager.scheduler.removeAtIndex(index)
            event.channel.sendMessage("Successfully removed the song from the queue!").queue()
        } catch (ex: NumberFormatException) {
            event.channel.sendMessage("$indexStr is not a number!").queue()
        }
    }
}